Modules/Server/Sources/PocketCastsServer/Private/TokenHelper.swift

```diff
@@ -105,4 +109,43 @@ class TokenHelper {
 
         return nil
     }
+
+    // MARK: - SSO Identity Token
+
+    private class func acquireIdentityToken() -> String? {
+        let semaphore = DispatchSemaphore(value: 0)
+        var refreshedToken: String? = nil
+        ApiServerHandler.shared.refreshIdentityToken { result in
+            switch result {
+            case .success(let token):
+                refreshedToken = token
+            case .failure:
+                refreshedToken = nil
+            }
+            semaphore.signal()
+        }
+        semaphore.wait()
```

Comment:
This follows the same Sync flow that the Password option uses. I would like to iterate this to track where the synchronous requirement originates from, but that would extend the scope of this change.